### ISSUE TITLE
TileLayer isLoaded when inside a composite Layer throws an exception

### ISSUE THREAD

**Issue by mogmog** [visgl/deck.gl#5165]
#### Description
When I use TileLayer on its own and check it's loaded state inside onAfterRender, i can inspect the isLoaded property.

i have encountered an issue where when I put this same tile layer inside a composite layer the tile layer internal state object is null at some point, meaning that isLoaded throws an exception when I inspect inside onAfterRender.

#### Repro Steps

see the following code pen - the normal tile layer can be swapped in by changing the layer prop to see that it works perfectly when not in a composite layer.

 https://codesandbox.io/s/youthful-chatelet-2b3ks?file=/index.js:1766-1779

#### Environment (please complete the following information):
 - Framework Version: 8.1.5
 - Browser Version: Chrome 86
 - OS: Mac OSX 10.14

#### Logs

Cannot read property 'tileset' of null

**Comment by Pessimistress** [comment]
Thanks for reporting. There is an internal lifecycle issue if a composite layer updates on viewport change. When `isLoaded` is called, `internalState.subLayers` are not the layers that have been rendered. They would in fact be added to the layer stack in the next render cycle.

While we address this, you can easily work around this by removing the `shouldUpdateState` method in your composite layer. You do not change any of the TileLayer props based on viewport, therefore there is no reason for the parent layer to rerender on every viewport change.

**Comment by mogmog** [comment]
this was a little bit of a contrived example. I do actually use shouldUpdateState to update layer state based on the viewport updates. Do you have any suggestions on a workaround for now? I can work around it for now so no worries if not.

**Comment by Pessimistress** [comment]
For now you can add this to your custom layer:

```js
  activateViewport(viewport) {
    const oldViewport = this.internalState.viewport;
    this.internalState.viewport = viewport;

    if (!oldViewport || !oldViewport.equals(viewport)) {
      this.setChangeFlags({ viewportChanged: true });
    }
  }
```

**Comment by Pessimistress** [comment]
Fix will be published in 8.4.0. You can try it with 8.4.0-alpha.
